.github/workflows/ci.yml

```diff
@@ -173,7 +183,70 @@ jobs:
           AZURE_CLIENT_ID: ${{ secrets.CLIENT_ID }}
           AZURE_CLIENT_SECRET: ${{ secrets.CLIENT_SECRET }}
           AZURE_TENANT_ID: ${{ secrets.TENANT_ID }}
-          CORSO_CI_TESTS: true 
+          CORSO_CI_TESTS: true
+          CORSO_M365_TEST_USER_ID: ${{ secrets.CORSO_M365_TEST_USER_ID }}
+          CORSO_PASSPHRASE: ${{ secrets.INTEGRATION_TEST_CORSO_PASSPHRASE }}
+        run: |
+          set -euo pipefail
+          go test \
+          -json   \
+          -v      \
+          ./... 2>&1 | tee ./testlog/gotest.log | gotestfmt -hide successful-tests
+
+      # Upload the original go test log as an artifact for later review.
+      - name: Upload test log
+        if: failure()
+        uses: actions/upload-artifact@v3
+        with:
+          name: test-log
+          path: src/testlog/gotest.log
+          if-no-files-found: error
+          retention-days: 14
+
+  Test-Suite-Fork:
+    needs: [Precheck]
+    environment: Testing
+    runs-on: ubuntu-latest
+    if:
+      github.event_name == 'repository_dispatch' &&
+      github.event.client_payload.slash_command.sha != '' &&
+      contains(github.event.client_payload.pull_request.head.sha, github.event.client_payload.slash_command.sha)
+    defaults:
+      run:
+        working-directory: src
+    steps:
+      # Check out merge commit
+      - name: Fork based /ok-to-test checkout
+        uses: actions/checkout@v2
+        with:
+          ref: 'refs/pull/${{ github.event.client_payload.pull_request.number }}/merge'
+
+      - name: Setup Golang with cache
+        uses: magnetikonline/action-golang-cache@v3
+        with:
+          go-version-file: src/go.mod
```

Comment:
@meain little confused here. Looks like we have the `Setup Golang with cache` step in the `Testing-Suite` jobs but also a step inside of `Checkout` to setup a golang cache. Do we need both?

Reply:
To be frank I had blindly copied it from the previous implementation by @ryanfkeepers . I think we will need to retain it here, don't think it gets transferred over between jobs.

Reply:
Each job runs in an isolated VM.  Caches you retrieved in initial jobs will not carry over to the later ones.  Similar to why we need to call the checkout action at the top of each job. 

On the topic, a note on the difference between the two: `./.github/actions/go-setup-cache` ensures we retrieve any modules missing from the cache, and will update the cache afterward (thus we only call it once, at the beginning).  `magnetikonline/action-golang-cache@v3` will read available caches, but won't update them.

Reply:
We recently merged a PR which enables separate caches for each step. We might wanna incorporate that changes here.
ref: https://github.com/alcionai/corso/pull/1532

Reply:
I see it in the PR here: https://github.com/alcionai/corso/blob/749dd6c1abc1f3818f2612de6ad97084b5bb2129/.github/workflows/ci.yml#L154

But it's not in main: https://github.com/alcionai/corso/blob/main/.github/workflows/ci.yml#L153

Am I missing something?

Reply:
We can skip that. Another follow up PR removed it the next day 🤣 .
https://github.com/alcionai/corso/pull/1534